sympy/assumptions/tests/test_query.py

```diff
@@ -1046,7 +1046,11 @@ def test_bounded():
 
     # exponential functions
     assert ask(Q.finite(log(x))) is None
-    assert ask(Q.finite(log(x)), Q.finite(x)) is True
+    assert ask(Q.finite(log(x)), Q.finite(x)) is None
+    assert ask(Q.finite(log(x)), Q.nonzero(x)) is None
```

Comment:
Does `nonzero` imply finite? Should this be true?

Reply:
`oo` is nonzero, but `log(oo)` is infinite so `nonzero` alone does not imply `log(x)` is finite

Reply:
In the old assumptions `nonzero` means `real & !zero` and `real` implies `finite`. So `oo` is not `nonzero`. That seems to be the same in the new assumptions as well:
```python
In [5]: oo.is_nonzero
Out[5]: False

In [6]: ask(Q.nonzero(oo))
Out[6]: False
```

Reply:
Oh, okay.

Reply:
I think the problem is that the facts in assumption modules are wrong. Old assumptions states that `complex` implies `finite`, but new assumption system does not...

Reply:
Fixed this.